metis-remove-data/src/main/java/eu/europeana/metis/remove/discover/AbstractOrphanIdentification.java

```diff
@@ -37,19 +38,29 @@
   enum DiscoveryMode {
 
     /**
-     * This mode removes all orphans and their descendants.
+     * This mode discovers only identified orphans or their descendants that have no children. The
```

Comment:
I'm not sure I understand the modes exactly, couldn't we just take into account deleted nodes as well anyway?
The deletion of them from the history or not is controlled from another script isn't it?

Reply:
The deletion is controlled from another script (in RemovePluginsMain). But we can execute this code in two ways. First, without deleting the history (what we're doing now). The second way is with also deleting history (which is what we did for instance when we want to completely delete a dataset because it is a duplicate). 

But if we also delete from the history, we have to ignore the DataStatus, because then we need to make sure that the node is not referenced from any other node as a child at all (so we would also delete nodes that have DataStatus DELETED. 

If we don't delete from the history, but rather mark as DELETED, then other plugins that are marked as DELETED are already 'done' and we don't have to process them anymore. 

So I think that there really is a need for this difference. Or am I missing something?

Reply:
(After our call this is cleared out)